src/script/modules/peeking-indicator/index.ts

```diff
@@ -0,0 +1,302 @@
+import settings from '../../lib/settings';
+import Module from '../../lib/module';
+import { getSnapchatStore } from '../../utils/snapchat';
+import styles from './index.module.css';
+import { logInfo } from '../../lib/debug';
+
+const store = getSnapchatStore();
+let initialized = false;
+
+class PeekingIndicator extends Module {
+  constructor() {
+    super('Peeking Indicator');
+    settings.on('HALF_SWIPE_NOTIFICATION.setting:update', this.load.bind(this));
+
+    this.load();
+  }
+
+  findSnapchatConversationContainers() {
+    const possibleContainers = [
+      ...document.querySelectorAll('.O4POs'),
+      ...document.querySelectorAll('[data-testid="conversation_item_container"]'),
+      ...document.querySelectorAll('[data-role="conversation_item"]'),
+      ...document.querySelectorAll('[role="listitem"]'),
+    ];
+    return possibleContainers;
+  }
+
+  initMutationObserver() {
+    if (initialized) return;
+
+    initialized = true;
+
+    const observer = new MutationObserver((mutations) => {
+      let shouldCheckForContainers = false;
+
+      mutations.forEach((mutation) => {
+        if (mutation.type === 'childList' && mutation.addedNodes.length > 0) {
+          shouldCheckForContainers = true;
+        }
+      });
+
+      if (shouldCheckForContainers) {
+        this.processConversationContainers();
+      }
+    });
+
+    observer.observe(document.body, {
+      childList: true,
+      subtree: true,
+    });
+
+    this.processConversationContainers();
+    setInterval(() => this.processConversationContainers(), 2000);
```

Comment:
Why does this need to be on an interval? doesn't this get called when the tree changes